skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -0,0 +1,267 @@
+package com.evolutiongaming.skafka.consumer
+
+import java.lang.{Long => LongJ}
+import java.time.Instant
+import java.util.{Map => MapJ}
+
+import cats.data.{NonEmptyMap => Nem, NonEmptySet => Nes}
+import cats.implicits._
+import cats.{Monad, ~>}
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.{ApplicativeThrowable, MonadThrowable, ToTry}
+import com.evolutiongaming.skafka.Converters._
+import com.evolutiongaming.skafka._
+import com.evolutiongaming.skafka.consumer.ConsumerConverters._
+import com.evolutiongaming.skafka.consumer.RebalanceCallback.Helpers._
+import org.apache.kafka.clients.consumer.{Consumer => ConsumerJ}
+import org.apache.kafka.common.{TopicPartition => TopicPartitionJ}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+// TODO review if we can do it without F[_] everywhere
```

Comment:
you can do that. just make `RebalanceCallback[+F[_], A]` and use `Nothing` in places without references to `F`